api/app/db/crud/weather_models.py

```diff
@@ -259,58 +259,45 @@ def get_latest_station_model_prediction_per_day(session: Session,
-                    station_code,
-                    date(prediction_timestamp) AS unique_day
-                FROM
-                    weather_station_model_predictions
-                WHERE
-                    station_code IN ({",".join(str(s) for s in station_codes)})
-                    AND prediction_timestamp >= '{day_start.isoformat()}'
-                    AND prediction_timestamp <= '{day_end.isoformat()}'
-                    AND date_part('hour', prediction_timestamp) = 20
-                GROUP BY
-                    station_code,
-                    unique_day) latest ON t.station_code = latest.station_code
-            JOIN (
-                SELECT
-                    prediction_model_run_timestamps.id,
-                    selected_model.abbreviation
-                FROM
-                    prediction_model_run_timestamps
-                    JOIN (
-                        SELECT
-                            id,
-                            abbreviation
-                        FROM
-                            prediction_models
-                        WHERE
-                            prediction_models.abbreviation = '{model}') selected_model 
-                            ON prediction_model_run_timestamps.prediction_model_id = selected_model.id) AS prediction_run_timestamps 
-                            ON t.prediction_model_run_timestamp_id = prediction_run_timestamps.id
-            AND date(t.prediction_timestamp) = latest.unique_day
-            AND t.prediction_timestamp = latest.latest_prediction
-        ORDER BY
-            update_date DESC;
-    """)
+    subquery = (
+        session.query(
+            func.max(WeatherStationModelPrediction.prediction_timestamp).label('latest_prediction'),
+            WeatherStationModelPrediction.station_code,
+            func.date(WeatherStationModelPrediction.prediction_timestamp).label('unique_day')
+        )
+        .filter(
+            WeatherStationModelPrediction.station_code.in_(station_codes),
+            WeatherStationModelPrediction.prediction_timestamp >= day_start,
+            WeatherStationModelPrediction.prediction_timestamp <= day_end,
+            func.date_part('hour', WeatherStationModelPrediction.prediction_timestamp) == 20
+        )
+        .group_by(
+            WeatherStationModelPrediction.station_code,
+            func.date(WeatherStationModelPrediction.prediction_timestamp).label('unique_day')
+        )
+        .subquery('latest')
+    )
+
+    result = session.query(
+        WeatherStationModelPrediction.id,
+        WeatherStationModelPrediction.prediction_timestamp,
+        PredictionModel.abbreviation,
+        WeatherStationModelPrediction.station_code,
+        WeatherStationModelPrediction.rh_tgl_2,
+        WeatherStationModelPrediction.tmp_tgl_2,
+        WeatherStationModelPrediction.bias_adjusted_temperature,
+        WeatherStationModelPrediction.bias_adjusted_rh,
+        WeatherStationModelPrediction.delta_precip,
```

Comment:
sorry I didn't catch this in your previous PR - `delta_precip` is only the precip that's occurred since the previous model interval. E.g., for HRDPS, it would only be the precip that's occurred in the last hour. What we want is `apcp_sfc_0` - that's the precip for the previous 24 hours.

Reply:
Good catch! Updated in: https://github.com/bcgov/wps/pull/2676/commits/151e041b913ea5c7f5915fa204c607c88b6e869d